core/src/main/java/org/hibernate/ogm/query/impl/OgmQueryTranslator.java

```diff
@@ -80,9 +80,17 @@
 	 */
 	private SelectClause selectClause;
 
+	/**
+	 * When the query is only targeting one type, we register the relevant information here.
+	 */
 	private EntityKeyMetadata singleEntityKeyMetadata;
 
 	/**
+	 * When the query is only targeting one type, we register the relevangt information here.
```

Comment:
Changed it to singleEntityTypeName.

There is not really a difference between the 2. It's just that the EntityKeyMetadata doesn't contain the type name and we require it so we have to have it there.

I considered adding it to EntityKeyMetadata following Emmanuel's comments but I'm not sure it's a good idea as it's part of the cache key.